index.js

```diff
@@ -202,18 +212,74 @@ module.exports = function leafletImage(map, callback) {
         im.crossOrigin = '';
 
         im.onload = function() {
+            if (loaded) return;
+            loaded = true;
             ctx.drawImage(this, x, y, size[0], size[1]);
             callback(null, {
                 canvas: canvas
             });
         };
 
-        im.src = url;
+        if (useAjax && !isBase64) {
+            if (cache === undefined) {
+                getBase64(url, function (data) {
+                    im.src = imageCache[marker._icon.src] = 'data: image/png;base64, ' + data;
+                    if (!loaded) {
+                        setTimeout(function() {
+                            im.onload();
+                        },0);
+                    }
+                });
+            } else {
+                im.src = cache;
+                if (!loaded) {
+                    setTimeout(function() {
+                        im.onload();
+                    },0);
+                }
+            }
+        } else {
+            im.src = url;
 
-        if (isBase64) im.onload();
+            if (isBase64 && !loaded) {
+                setTimeout(function () {
```

Comment:
There is always that chance that it had something to do with the gravitation and alignment of the planets...  I'll try it later without it just to make sure.

:P